# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsTouchBar.mm
# Commit: 6f13ec193094
# Full Hash: 6f13ec193094a7c7507edd8df79fdb0e1c985c40
# Author: harry <htwyford@mozilla.com>
# Date: 2019-10-11 21:50:12
# Description:
#   Bug 1588084 - Add null-checking to isUrlbarFocused to avoid crashes on window shutdown. r=mikedeboer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D48966
# ==============================================================================

diff -r f3f4aa35d46e -r 6f13ec193094 widget/cocoa/nsTouchBar.mm
--- a/widget/cocoa/nsTouchBar.mm	Thu Oct 10 18:51:45 2019 +0000
+++ b/widget/cocoa/nsTouchBar.mm	Fri Oct 11 13:35:04 2019 +0000
@@ -362,6 +362,11 @@
 
   // Special handling to show/hide the search popover if the Urlbar is focused.
   if ([[aInput nativeIdentifier] isEqualToString:SearchPopoverIdentifier]) {
+    // We can reach this code during window shutdown. We only want to toggle
+    // showPopover if we are in a normal running state.
+    if (!mTouchBarHelper) {
+      return;
+    }
     bool urlbarIsFocused = false;
     mTouchBarHelper->GetIsUrlbarFocused(&urlbarIsFocused);
     if (urlbarIsFocused) {
